src/Tribe/Views/V2/iCalendar/Links/Link_Abstract.php

```diff
@@ -211,27 +234,50 @@ public function get_label( View $view = null ) {
 		 *
 		 * @return string $label The label that will be displayed.
 		 */
-		return apply_filters( "tec_views_v2_subscribe_links_{$slug}_label", $this->label, $this, $view );
+		return (string) apply_filters( "tec_views_v2_single_subscribe_links_{$slug}_label", $value, $this, $view );
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
-	public function get_single_label( View $view = null ) {
+	public function get_single_label( View $view = null ): string {
+		return $this->filter_get_single_label( $this->single_label(), $view );
+	}
+
+	/**
+	 * Returns the label for the single event view.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	abstract protected function single_label(): string;
+
+	/**
+	 * Filters the single label for the link.
+	 *
+	 * @since TBD
+	 *
+	 * @param string    $value The label to filter.
+	 * @param View|null $view  The current View object.
+	 *
+	 * @return string
+	 */
+	protected function filter_get_single_label( string $value, View $view = null ): string {
 		$slug = self::get_slug();
 
 		/**
 		 * Allows filtering of the labels for the Single Event view labels.
 		 *
 		 * @since 5.12.0
 		 *
-		 * @param string        $label    The label that will be displayed.
-		 * @param Link_Abstract $link_obj The link object the label is for.
-		 * @param View          $view     The current View object.
+		 * @param string        $single_label The label that will be displayed.
+		 * @param Link_Abstract $link_obj     The link object the label is for.
+		 * @param View          $view         The current View object.
 		 *
 		 * @return string $label The label that will be displayed.
 		 */
-		return apply_filters( "tec_views_v2_single_subscribe_links_{$slug}_label", $this->single_label, $this, $view );
+		return (string) apply_filters( "tec_views_v2_single_subscribe_links_{$slug}_single_label", $value, $this, $view );
```

Comment:
Deprecating instead of directly changing ?

Reply:
Why deprecate? It's the same filter

Reply:
Its not, original was `_label` while new is `_single_label`